src/cryptography/hazmat/backends/openssl/dh.py

```diff
@@ -180,3 +206,17 @@ def public_numbers(self):
 
     def parameters(self):
         return _dh_cdata_to_parameters(self._dh_cdata, self._backend)
+
+    def public_bytes(self, encoding, format):
+        if format is serialization.PublicFormat.PKCS1:
```

Comment:
The error handling in `_public_key_bytes` won't handle the case where you pass openssh for both format and encoding (you'll get an assertion error because the openssh code path assumes the fallthrough is elliptic curve). Since we only support one publicformat here I think just doing `if format not serialization.PublicFormat.SubjectPublicKeyInfo: raise` is probably a reasonable path.